test/pkg/steps/kie_asset_library.go

```diff
@@ -0,0 +1,158 @@
+
+	cloneOptions := &git.CloneOptions{
+		URL:          KieAssetReMarshallerGitRepositoryURI,
+		SingleBranch: true,
+	}
+
+	var err error
+	reference := KieAssetReMarshallerGitRepositoryBranch
+	if len(reference) == 0 {
+		err = cloneRepository(data.KieAssetReMarshallerLocation, cloneOptions)
+	} else {
+		// Try cloning as branch reference
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(reference)
+		err = cloneRepository(data.KieAssetReMarshallerLocation, cloneOptions)
+		// If branch clone was successful then return, otherwise try other cloning options
+		if err == nil {
+			return nil
+		}
+
+		// If branch cloning failed then try cloning as tag
+		cloneOptions.ReferenceName = plumbing.NewTagReferenceName(reference)
+		err = cloneRepository(data.KieAssetReMarshallerLocation, cloneOptions)
+	}
+	return err
+}
+
+func (data *Data) projectKieAssetLibraryIsCloned() error {
+	framework.GetLogger(data.Namespace).Info("Cloning kie-asset-library project", "URI", KieAssetLibraryGitRepositoryURI, "branch", KieAssetLibraryGitRepositoryBranch, "clonedLocation", data.KieAssetLibraryLocation)
+
+	cloneOptions := &git.CloneOptions{
+		URL:          KieAssetLibraryGitRepositoryURI,
+		SingleBranch: true,
+	}
+
+	var err error
+	reference := KieAssetLibraryGitRepositoryBranch
+	if len(reference) == 0 {
+		err = cloneRepository(data.KieAssetLibraryLocation, cloneOptions)
+	} else {
+		// Try cloning as branch reference
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(reference)
+		err = cloneRepository(data.KieAssetLibraryLocation, cloneOptions)
+		// If branch clone was successful then return, otherwise try other cloning options
+		if err == nil {
+			return nil
+		}
+
+		// If branch cloning failed then try cloning as tag
+		cloneOptions.ReferenceName = plumbing.NewTagReferenceName(reference)
+		err = cloneRepository(data.KieAssetLibraryLocation, cloneOptions)
+	}
+	return err
+}
+
+func (data *Data) projectIsGeneratedByKieAssetLibrary(project string) error {
+	if _, err := os.Stat(data.KieAssetLibraryLocation + "/kie-assets-library-generate/target/" + project); !os.IsNotExist(err) {
+		return err
+	}
+	return nil
+}
+
+func (data *Data) projectInKieAssetLibraryIsBuiltByMaven(project string) error {
+	output, errCode := framework.CreateMavenCommand(data.KieAssetLibraryLocation+"/kie-assets-library-generate/target/"+project).
+		SkipTests().
+		Execute("clean", "install")
+	framework.GetLogger(data.Namespace).Debug(output)
+	if errCode != nil {
+		framework.GetLogger(data.Namespace).Warn(project + " 'mvn clean install' failed due to: " + errCode.Error())
+	}
+	return errCode
+}
+
+func (data *Data) projectAssetsAreRemarshalledByVsCode(project string) error {
+	installOutput, installErr := framework.CreateCommand("npm", "install").
+		WithRetry(framework.NumberOfRetries(1)).
+		InDirectory(data.KieAssetReMarshallerLocation).
+		Execute()
+	framework.GetLogger(data.Namespace).Debug(installOutput)
```

Comment:
display output only in error or?

Reply:
I was asked earlier to keep debug log message for such commands, see https://github.com/kiegroup/kogito-operator/pull/860#pullrequestreview-651149860

Reply:
I guess we should "debug" if no error but if error occurs, we should still put it as error log I guess